circuitClient.js

```diff
@@ -123,6 +125,61 @@ class CircuitClient {
     });
   }
 
+  async setPresenceAvailable() {
+    const statusMsg = await this.client.getStatusMessage();
+    try {
+      return this.client.setPresence({
+        state: Circuit.Enums.PresenceState.AVAILABLE,
+        statusMessage: statusMsg
+      });
+    } catch {
+        console.log("Could not set user to available");
+    }
+  }
+  
+  async setPresenceDnd(untilTime, duration) {
```

Comment:
Actually reading this function again I don't think it needs to be an async function, since you are returning promises. So if in the case of the final else you could return Promise.reject